app/javascript/channels/room_channel.js

```diff
@@ -0,0 +1,40 @@
+import consumer from './consumer'
+
+function formatDate(date) {
+    return date.getFullYear() + "-" + ("0" + (date.getMonth() + 1)).slice(-2) + "-" + ("0" + date.getDate()).slice(-2) +
```

Comment:
can't you somehow use the method on the message model?

Reply:
Explain how

Reply:
Technically we could transform the date in the model JSON already, such that the frontend only has to take that and display it. Our thought behind the current version was to keep a unified output from the backend and let the frontend decide on how to display the date.

Reply:
In `spec/features/global_chat_spec.rb` you use the method from the model to check whether the timestamp is displayed correctly and in `views/room_messages/_message.html.erb` you use the method to display the timestamp. So I think you should use the same method when a new message arrives. I think you could use the `as_json` method of the model as already said by Tobias.

Reply:
You are already doing this for the `display_name` so it should be quite easy ;)

Reply:
And for maintainability this could also be considered for the link, so one has not so many places to change it if it ever changes :sweat_smile:

Reply:
Fixed in
f0be623